homeassistant/components/samsungtv/__init__.py

```diff
@@ -304,9 +304,26 @@ async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
                 new_connections = device.connections.copy()
                 new_connections.discard((dr.CONNECTION_NETWORK_MAC, "none"))
                 if new_connections != device.connections:
-                    dev_reg.async_update_device(
-                        device.id, new_connections=new_connections
+                    # The migration is failing for some users, so we log the error
+                    # see #119082 for sample errors
+                    LOGGER.info(
+                        "Clearing invalid `none` mac connection "
+                        "from device %s for config entry %s",
+                        device.id,
+                        config_entry.entry_id,
                     )
+                    try:
+                        dev_reg.async_update_device(
+                            device.id, new_connections=new_connections
+                        )
+                    except KeyError:
```

Comment:
I was unable to replicate, so I am unable to add tests and coverage failure for this is expected.